### ISSUE TITLE
New lint: Implement From instead of Into

### ISSUE THREAD

**Issue by davidpdrsn** [rust-lang/rust-clippy#6456]
### What it does

Finds implementations of `Into` and suggests implementing `From` instead. That should always be possible since Rust 1.41 as per https://doc.rust-lang.org/stable/std/convert/trait.Into.html#implementing-into-for-conversions-to-external-types-in-old-versions-of-rust.

According the std docs implementing `From` is preferred since it gives you `Into` for free where the reverse isn't true:

> It is important to understand that Into does not provide a From implementation (as From does with Into). Therefore, you should always try to implement From and then fall back to Into if From can't be implemented.

### Categories (optional)

- Kind: Style.

### Drawbacks

There might still be cases where one cannot implement `From` but implementing `Into` is possible. If such cases exist and are hard to detect this lint might not be a good idea. Could lead to false positives.

### Example

```rust
struct StringWrapper(String);

impl Into<StringWrapper> for String {
    fn into(self) -> StringWrapper {
        StringWrapper(self)
    }
}
```

Could be written as:

```rust
struct StringWrapper(String);

impl From<String> for StringWrapper {
    fn from(s: String) -> StringWrapper {
        StringWrapper(s)
    }
}
```

**Comment by giraffate** [comment]
related issue: https://github.com/rust-lang/rust-clippy/issues/1567

**Comment by 1c3t3a** [comment]
I would like to work on this!
